packages/medusa-react/src/hooks/admin/customer-groups/queries.ts

```diff
@@ -66,3 +68,24 @@ export const useAdminCustomerGroups = (
   )
   return { ...data, ...rest } as const
 }
+
+/**
+ * Hook retrieves a list of customers that belong to provided groups.
+ *
+ * @param id - customer group id
+ * @param query - pagination/filtering params
+ * @param options
+ */
+export const useAdminCustomerGroupCustomers = (
+  id: string,
+  query?: AdminGetCustomersParams,
+  options?: UseQueryOptionsWrapper<Response<AdminCustomersListRes>, Error, any>
+) => {
+  const { client } = useMedusa()
+  const { data, ...rest } = useQuery(
+    [...adminCustomerGroupKeys.detail(id), "customers", query],
+    () => client.admin.customerGroups.listCustomers(id, query),
+    options
+  )
+  return { ...data, ...rest } as const
+}
```

Comment:
nit: would be cool to be able to extend the `adminCustomerGroupKeys` to include a method that returns the above query key. I'm thinking something along those lines:

```typescript
const baseQueryKeys = queryKeysFactory(ADMIN_CUSTOMER_GROUPS_QUERY_KEY)

export const adminCustomerGroupKeys = {
  ...baseQueryKeys,
  detailCustomer: (id: string, query?: AdminGetCustomersParams) =>
    [
      ...baseQueryKeys.detail(id),
      "customers",
      query,
    ] as const,
}
```

Then you'd also be able to replace the `options` type with this:

```diff
export const useAdminCustomerGroupCustomers = (
  id: string,
  query?: AdminGetCustomersParams,
  options?: UseQueryOptionsWrapper<
    Response<AdminCustomersListRes>,
    Error,
-   any
+   ReturnType<CustomerGroupQueryKeys["detailCustomer"]>
  >
) => {...}
```

Let me know what you think 😄 !

Reply:
I had a similar idea initially (to provide the extend param to `queryKeysFactory`) but I like the approach of extending `adminCustomerGroupKeys` directly even more, thanks 👍